File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return root;

        Deque<TreeNode> myStack = new LinkedList<>();
        
        myStack.push(root);

        while (myStack.size() > 0) {
            TreeNode curNode = myStack.pop();

            TreeNode tmp = curNode.left;
            curNode.left = curNode.right;
            curNode.right = tmp;

            if (curNode.left != null)
                myStack.push(curNode.left);

            if (curNode.right != null)
                myStack.push(curNode.right);
        }

        return root;
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        if root.left is None and root.right is None:
            return 0
        if root.left is None:
            return max(self.maxDepth(root.right),self.diameterOfBinaryTree(root.right))
        if root.right is None:
            return max(self.maxDepth(root.left),self.diameterOfBinaryTree(root.left))
        return self.maxDepth(root.right) + self.maxDepth(root.left)

    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        else:
            return 1 + max(self.maxDepth(root.right), self.maxDepth(root.left))